src/document_basic.c

```diff
@@ -167,21 +167,21 @@ int Document_LoadSchemaFieldJson(Document *doc, RedisSearchCtx *sctx) {
   doc->score = SchemaRule_JsonScore(sctx->redisCtx, rule, jsonKey, keyName);
   // No payload on JSON as RedisJSON does not support binary fields
 
-  JSONType type;
+  JSONType jsonType;
   doc->fields = rm_calloc(nitems, sizeof(*doc->fields));
```

Comment:
Do we need to check for an existing `doc->fields` (and deallocate it) before assigning a new value here?